exp/operator/controllers/coreprovider_controller_test.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	operatorv1 "sigs.k8s.io/cluster-api/exp/operator/api/v1alpha4"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ reconcile.Reconciler = &CoreProviderReconciler{}
+
+var _ = Describe("CoreProvider Reconciler", func() {
```

Comment:
We agreed on using go test framework instead of ginkgo, see https://cluster-api.sigs.k8s.io/developer/testing.html for more context

Reply:
> I used the pattern that I saw here https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/machineset_controller_test.go#L42

I'm sorry you took as an example an test that is not yet aligned on the current recommendations.

> Are there plans to change other tests too?

Yep, /priority backlog /kind cleanup since the new recommendation are in place, but we see some progress, e.g. https://github.com/kubernetes-sigs/cluster-api/pull/4451